### PR TITLE
Changelog: prepare for the release of version 2.0.0

### PR DESCRIPTION
**Pull request by enricobattocchi**
The release date has been tentatively set to next Thursday, December 28, 2023.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,21 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## [2.0.0] - 2023-12-28
+### Changed
+* All the source classes are now namespaced under `Yoast\WHIPv2`. The version number in the namespaced will be bumped up with every major version. [#157]
+    The classes have also been renamed to remove the `Whip_` prefix, and the folders' names have been capitalized to follow the PSR-4 standard.
+* The `Requirement` interface now declares explicitly two methods: `version() ` and `operator()`. [#146]
```
Comment (jrfnl) [inline CHANGELOG.md:11]:
```suggestion
* The `Requirement` interface now explicitly declares the following two additional methods: `version() ` and `operator()` and classes implementing the interface should ensure these methods are available. [#146]
```

**Review by jrfnl — COMMENTED** [review]
@enricobattocchi Looks good.

I wonder if the following things should also be mentioned:
* That the plugin is now tested up to PHP 8.3.
* PR #133 - Improved PHP 8.1 compatibility (PRs #144 and #145 are follow ups on this and fix new bugs introduced by #133, so don't need a explicit mention).
* PR #158 - Removed method.

**Comment by enricobattocchi** [comment]
>     * That the plugin is now tested up to PHP 8.3.

Right! Pushed.

>     * PR [132 whip php 81 improvements #133](https://github.com/Yoast/whip/pull/133) - Improved PHP 8.1 compatibility (PRs [Whip_MessageDismisser::verifyNonce(): bug fix #144](https://github.com/Yoast/whip/pull/144) and [Whip_WPMessageDismissListener::listen(): bug fix #145](https://github.com/Yoast/whip/pull/145) are follow ups on this and fix new bugs introduced by [132 whip php 81 improvements #133](https://github.com/Yoast/whip/pull/133), so don't need a explicit mention).

#132 is not actually about PHP 8.1 compatibility I think? We removed the undesirable `filter_input` occurrences but they weren't actually deprecated since they weren't using any `FILTER_SANITIZE_*`... so maybe it's not worth mentioning?

>     * PR [Whip_WPMessagePresenter: remove deprecated method #158](https://github.com/Yoast/whip/pull/158) - Removed method.

Right, pushed.

**Comment by jrfnl** [comment]
> > * PR [132 whip php 81 improvements #133](https://github.com/Yoast/whip/pull/133) - Improved PHP 8.1 compatibility (PRs [Whip_MessageDismisser::verifyNonce(): bug fix #144](https://github.com/Yoast/whip/pull/144) and [Whip_WPMessageDismissListener::listen(): bug fix #145](https://github.com/Yoast/whip/pull/145) are follow ups on this and fix new bugs introduced by [132 whip php 81 improvements #133](https://github.com/Yoast/whip/pull/133), so don't need a explicit mention).
> 
> #132 is not actually about PHP 8.1 compatibility I think? We removed the undesirable `filter_input` occurrences but they weren't actually deprecated since they weren't using any `FILTER_SANITIZE_*`... so maybe it's not worth mentioning?

Well, maybe it is worth mentioning as security hardening as no filter was being passed previously, which meant that `FILTER_DEFAULT` was used, which basically meant that the input was not being sanitized.... ?

**Comment by enricobattocchi** [comment]
> Well, maybe it is worth mentioning as "security hardening" as no filter was being passed previously, which meant that `FILTER_DEFAULT` was used, which basically meant that the input was not being sanitized.... ?

Makes total sense, pushed

**Review by jrfnl — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**